thinc/tests/layers/test_layers_api.py

```diff
@@ -128,6 +128,7 @@ def assert_data_match(Y, out_data):
     # ("CauchySimilarity.v1", {}, (array2d, array2d), array1d),
     ("ParametricAttention.v1", {}, ragged, ragged),
     ("SparseLinear.v1", {}, (numpy.asarray([1, 2, 3], dtype="uint64"), array1d, numpy.asarray([1, 1], dtype="i")), array2d),
+    ("SparseLinear.v2", {}, (numpy.asarray([1, 2, 3], dtype="uint64"), array1d, numpy.asarray([1, 1], dtype="i")), array2d),
```

Comment:
Can we think of a specific test that would have caught this bug, though?

Reply:
I was thinking about this. Say that we have a sparselinear of a certain size and a certain variety of inputs, we can compute the expected number of non-zero values. Of course, it won't be exact since the hash is a randomization and there are collisions. But if the actual number of non-zeros is within a certain bandwidth (I guess the number of non-zeros are normally distributed), then the test would pass. We could do this for the rows of very class.

Hope that makes sense?

Reply:
Added a test that does this.